animating-widgets.md

```diff
@@ -0,0 +1,293 @@
+    duration: const Duration(milliseconds: 5000)
+  );
+  final animation = new Tween(
+    begin: 0.0,
+    end: 200.0)
+  .animate(controller);
+  runApp(
+    new MaterialApp(
+      title: 'My Animation Test',
+      routes: <String, WidgetBuilder> {
+        '/': (BuildContext context) => new LogoApp(
+          animation: animation
+        )
+      }
+    )
+  );
+  controller.forward();
+}
+```
+
+For more complex animations involving additional state, consider using the
+[`AnimatedBuilder`](http://docs.flutter.io/flutter/material/AnimatedBuilder-class.html)
+class.
+
+## Render simultaneous widget animations
+
+The following example shows how you can apply two simultaneous animation
+effects: one that controls the size of the widget and another that controls
+the opacity to fade the widget in and out.
+
+You can use the
+[`addStatusListener()`](http://docs.flutter.io/flutter/animation/Animation/addStatusListener.html)
+method to notify your widget when the current status of an animation changes.
+In this example, we use the status listener to switch the animation’s playback
+between going forward or reverse.
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:flutter/animation.dart';
+
+class LogoApp extends AnimatedWidget {
+
+  // Declare the Tween objects as static because we don't change them.
+  static final _opacityTween = new Tween<double>(begin: 1.0, end: 0.3);
+  static final _sizeTween = new Tween<double>(begin: 0.0, end: 300.0);
+
+  LogoApp({Key key, Animation<double> animation})
+      : super(key: key, animation: animation);
+
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Opacity(
+        opacity: _opacityTween.evaluate(animation),
+        child: new Container(
+          margin: new EdgeInsets.symmetric(vertical: 10.0),
+          height: _sizeTween.evaluate(animation),
+          width: _sizeTween.evaluate(animation),
+          child: new AssetImage(
+            name: 'assets/flutter_logo.png',
+            fit: ImageFit.contain
+          )
+        )
+      )
+    );
+  }
+}
+
+void main() {
+  final AnimationController controller = new AnimationController(
+    duration: const Duration(milliseconds: 5000)
+  );
+
+  final Animation<double> animation =
+      new CurvedAnimation(parent: controller, curve: Curves.bounceIn);
+
+  animation.addStatusListener((status) {
+    if (status == AnimationStatus.completed)
+      controller.reverse();
+    else if (status == AnimationStatus.dismissed)
+      controller.forward();
+  });
+
+  runApp(
+    new MaterialApp(
+      title: 'My Animation Test',
+      routes: <String, WidgetBuilder> {
+        '/': (BuildContext context) => new LogoApp(
+          animation: animation
+        )
+      }
+    )
+  );
+  controller.forward();
+}
+```
+
+## Dispose your widget animation
+
+It’s good practice to dispose your animation controller and free up your
+resources when they are no longer needed. To do so, call the
```

Comment:
It's more than a good practice.  It's required for your code to be correct and you'll now get an assert if you don't do it.